enterprise/app/code/code.tsx

```diff
@@ -0,0 +1,383 @@
+
+    requestingReview: false,
+  };
+
+  editor: any;
+
+  codeViewer = React.createRef<HTMLDivElement>();
+
+  octokit: any;
+
+  subscription: Subscription;
+
+  componentWillMount() {
+    document.title = `Code | BuildBuddy`;
+
+    this.octokit = new MyOctokit({
+      auth: this.props.user.selectedGroup.githubToken,
+    });
+
+    this.fetchCode();
+
+    this.subscription = rpcService.events.subscribe({
+      next: (name) => name == "refresh" && this.fetchCode(),
+    });
+  }
+  handleWindowResize() {
+    this.editor?.layout();
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", () => this.handleWindowResize());
+    // TODO(siggisim): select default file based on url
+    this.editor = monaco.editor.create(this.codeViewer.current, {
+      value: ["// Welcome to BuildBuddy Code!", "", "// Click on a file to the left to get start editing."].join("\n"),
+      theme: "vs",
+    });
+
+    this.editor.onDidChangeModelContent(() => {
+      this.handleContentChanged();
+    });
+  }
+
+  handleContentChanged() {
+    if (this.state.originalFileContents == this.editor.getValue()) {
+      this.state.changes.delete(this.state.currentFilePath);
+      this.state.pathToIncludeChanges.delete(this.state.currentFilePath);
+    } else if (this.state.currentFilePath) {
+      if (!this.state.changes.get(this.state.currentFilePath)) {
+        this.state.pathToIncludeChanges.set(this.state.currentFilePath, true);
+      }
+      this.state.changes.set(this.state.currentFilePath, this.editor.getValue());
+    }
+    this.setState({ changes: this.state.changes });
+    console.log(this.state.changes);
+    // TODO(siggisim): serialize these changes to localStorage or to server and pull them back out.
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", () => this.handleWindowResize());
+    this.subscription?.unsubscribe();
+    this.editor?.dispose();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.hash !== prevProps.hash ||
+      this.props.search != prevProps.search ||
+      this.props.path != this.props.path
+    ) {
+      this.fetchCode();
+    }
+  }
+
+  fetchCode() {
+    let groups = this.props.path?.match(/\/code\/(?<owner>.*)\/(?<repo>.*)/)?.groups;
+    if (!groups?.owner || !groups.repo) {
+      this.handleRepoClicked();
+      return;
+    }
+    this.setState({ owner: groups?.owner || "buildbuddy-io", repo: groups?.repo || "buildbuddy" }, () => {
+      this.octokit.request(`/repos/${this.state.owner}/${this.state.repo}/git/trees/master`).then((response: any) => {
+        console.log(response);
+        this.setState({ repoResponse: response });
+      });
+    });
+  }
+
+  // TODO(siggisim): Support deleting files
+  // TODO(siggisim): Support moving files around
+  // TODO(siggisim): Support renaming files
+  // TODO(siggisim): Support right click file context menus
+  // TODO(siggisim): Support tabs
+  handleFileClicked(node: any, fullPath: string) {
```

Comment:
nit: is there a type that could be imported instead of `any` here?

Reply:
I've added a todo to remove the use of all any types, here and elsewhere in the file